### DISCUSSION TITLE
Creating tag descriptions

### DISCUSSION THREAD

**Discussion by dryan-cais** [microsoft/typespec#3777] [OpenAPI3]
First off thanks for the development effort! 

We are current accessing Typespec as a means of better managing our 9000 line OpenAPI spec. It is looking very promising but we are missing the ability to add tag descriptions in the top level tags section. I don't think we will be able to move over to Typespec if it cannot generate our current schema.

I had a look at the code, and it doesn't look like this is currently supported? Is this on the roadmap? I wouldn't mind having a go at producing a PR, but I fear I might need some advice to get started as I am both unfamiliar with the codebase, and haven't programmed in TS recently. If someone can point me in the right direction in regards what files need to change, then I can give it a go.

Thanks again.

**Comment by timotheeguerin** [comment]
We’ve had this issue on the backlog for a while feel free to thumbs up, it has been helpful to see what the community wants to see prioritized
https://github.com/microsoft/typespec/issues/2220

the proposal in there with ‘@tagMetadata decorator is probably a good approach for contributions too

**Reply by dryan-cais** [reply]
That's awesome! Sorry I did try to search for any outstanding issues but didn't see this. Once we have completed moving over to Typespec, i'll try and create a PR for this.

**Comment by pschaeflein** [comment]
I can see that the @tagMetadata decorator is shipped. The OpenApi emitter processes it as expected.

However, the tsp compiler (and the language service in VSCode) reports an error. Is there an example/sample that I haven't found?

![image](https://github.com/user-attachments/assets/ba226441-2600-4e2c-82f6-165065323d92)

**Reply by timotheeguerin** [reply]
This works fine in the [playground](https://typespec.io/playground/?c=aW1wb3J0ICJAdHlwZXNwZWMvb3BlbmFwaSI7Cgp1c2luZyBPcGVuQVBJOwoKQHNlcnZpY2UKQHRhZ01ldGFkYXRhKAogICJUYWcgTmFtZSIsCiAgI3sKICAgIGRlc2NyaXB0aW9uOsYiyxLFKSAgZXh0ZXJuYWxEb2NzOsg5ICB1cmw6ICJodHRwczovL2V4YW1wbGUuY29txzfQXU1vcmUgaW5mby7JIWB4LWN1c3RvbWA6ICJzdHJpbmfHHH3GB9ghfQopCm5hbWVzcGFjZSBQZXRTdMReewoKfQo%3D&e=%40typespec%2Fopenapi3&options=%7B%7D) (and me locally) 

are you on the latest version of the compiler adn its libraries? Does it change if you move `@service` above?

**Reply by pschaeflein** [reply]
tsp --version reports   0.64.0

Running `tsp compile` does not show errors.

vscode extension is 0.64.1

The errors show in the problems panel only when the file containing the decorator is opened.
The errors show in the terminal window for task. (The auto-discovered task - in the "contributed" section of the VSCode command bar)

Order of the tags does not matter - same result...

**Reply by allenjzhang** [reply]
@pschaeflein , could you [create an bug issue](https://github.com/microsoft/typespec/issues/new/choose) . @RodgeFu for FYI.

**Reply by pschaeflein** [reply]
@allenjzhang - Will do. Thanks.

**Reply by timotheeguerin** [reply]
@pschaeflein just in case but did you try restarting the vscode extension/vscode itself?

And do you know which version of the openapi package you are using?

**Reply by pschaeflein** [reply]
>  restarting the vscode extension/vscode itself?

Not sure what you mean...

>  which version of the openapi package you are using?

The package.json in node_modules/@typespec/openapi has version 0.64.0 (Is that what you're asking?)

**Reply by timotheeguerin** [reply]
> The package.json in node_modules/@typespec/openapi has version 0.64.0 (Is that what you're asking?)

yes, so most likely not the issue

> restarting the vscode extension/vscode itself?

Did you try restarting VSCode? Or the reloading the TypeSpec language server `CtrlOrCmd+shift+p > Restart TypeSpec Language Server`

**Reply by pschaeflein** [reply]
Just restarted. Did not resolve.

**Reply by RodgeFu** [reply]
Thanks for the feedback, @pschaeflein . The error in vscode is not expected. We will investigate into the error in vscode and get back to you. thanks.

**Reply by randreu28** [reply]
I couldn't find an issue for this, so I created it

https://github.com/microsoft/typespec/issues/7468

**Reply by RodgeFu** [reply]
#5933 has been opened to track the issue.  Thanks.
